### PR TITLE
TA v3.1-preview.4 - updated examples and definitions for TA for Health

### PR DESCRIPTION
**Pull request by ctufts**
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i>

### Changelog
Please ensure to add changelog with this PR by answering the following questions.
  1. What's the purpose of the update?    
      - [ ] new service onboarding 
      - [ ] new API version 
      - [ ] update existing version for new feature 
      - [ ] update existing version to fix swagger quality issue in s360
      - [ ] Other, please clarify 
  2. When you are targeting to deploy new service/feature to public regions? Please provide date, or month to public if date is not available yet.
  3. When you expect to publish swagger? Please provide date, or month to public if date is not available yet.
  4. If it's an update to existing version,  please select SDKs of specific language and CLIs that require refresh after swagger is published.
      - [ ] SDK of .NET (need service team to ensure code readiness)
      - [ ] SDK of Python
      - [ ] SDK of Java
      - [ ] SDK of Js
      - [ ] SDK of Go
      - [ ] PowerShell
      - [ ] CLI
      - [ ] Terraform
      - [ ] No, no need to refresh for updates in this PR

### Contribution checklist:
- [ ] I commit to follow the [Breaking Change Policy](http://aka.ms/bcforapi) of “no breaking changes
- [ ] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [ ] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist
- [ ] Ensure to check this box if one of the following scenarios meet updates in the PR, so that label “WaitForARMFeedback” will be added automatically to involve ARM API Review. Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs, all “removals” and “adding a new property” no more require ARM API review.
  - Adding new API(s)
  - Adding a new API version
  - Adding a new service

- [ ] Please ensure you've reviewed following [guidelines](https://aka.ms/rpguidelines) including [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md). Estimated time (4 hours). This is required before you can request review from ARM API Review board.

- [ ] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them. 

### Breaking Change Review Checklist 
If there are following updates in the PR, ensure to request an approval from API Review Board as defined in the [Breaking Change Policy](http://aka.ms/bcforapi). 

- [ ] Removing API(s) in stable version
- [ ] Removing properties in stable version
- [ ] Removing API version(s) in stable version
- [ ] Updating API in stable version with Breaking Change Validation errors
- [ ] Updating API(s) in preview over 1 year

**Action**: to initiate an evaluation of the breaking change, create a new intake using the [template for breaking changes](https://aka.ms/Breakingchangetemplate). Addition details on the process and office hours are on the [Breaking change Wiki](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37684/Breaking-Changes).

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

**Review by tjprescott — APPROVED** [review]

**Comment by tjprescott** [comment]
@ctufts is this ready to merge?

**Comment by ctufts** [comment]
> @ctufts is this ready to merge?
@tjprescott  Not yet, I'd like to get a review in from some other TA members first. Will update you when it is ready for merging.

File: specification/cognitiveservices/data-plane/TextAnalytics/preview/v3.1-preview.4/TextAnalytics.json [hunk 1]
```diff
@@ -1774,12 +1774,14 @@
...
           "properties": {
-            "isNegated": {
-              "type": "boolean"
+            "assertions": {
+              "type": "object",
+              "$ref": "#/definitions/HealthcareAssertion"
+            },
+            "name": {
+              "description": "Preferred name for the entity. Example: 'histologically' would have a 'name' of 'histologic'.",
+              "type": "string"
```
Comment (assafi) [inline specification/cognitiveservices/data-plane/TextAnalytics/preview/v3.1-preview.4/TextAnalytics.json:1784]: use enum as the set of values is a closed set

Comment (ctufts) [reply specification/cognitiveservices/data-plane/TextAnalytics/preview/v3.1-preview.4/TextAnalytics.json:1784]: Not sure I understand - the name isnt' an enum/set of closed values

File: specification/cognitiveservices/data-plane/TextAnalytics/preview/v3.1-preview.4/TextAnalytics.json [hunk 2]
```diff
@@ -1796,25 +1798,67 @@
...
           "description": "Type of relation. Examples include: `DosageOfMedication` or 'FrequencyOfMedication', etc.",
           "type": "string"
         },
-        "bidirectional": {
-          "description": "If true the relation between the entities is bidirectional, otherwise directionality is source to target.",
-          "type": "boolean"
-        },
-        "source": {
-          "description": "Reference link to the source entity.",
+        "entities":{
```
Comment (assafi) [inline specification/cognitiveservices/data-plane/TextAnalytics/preview/v3.1-preview.4/TextAnalytics.json:1808]: We should add a `description` field for the `HealthcareRelation` type. Explaining that every relation is an entity graph of a certain `entityType`, where all entities are connected and have specific roles within the relation context.

File: specification/cognitiveservices/data-plane/TextAnalytics/preview/v3.1-preview.4/TextAnalytics.json [hunk 3]
```diff
@@ -1796,25 +1798,67 @@
...
+          "type": "string",
+          "enum": [
+            "Hypothetical",
+            "Conditional"
+          ]
+        },        
+        "certainty": {
+          "description": "Describes the entities certainty and sentiment. Possible values: positive, positive possible, neutral, negative possible, negative",
+          "type": "string",
+          "enum": [
```
Comment (assafi) [inline specification/cognitiveservices/data-plane/TextAnalytics/preview/v3.1-preview.4/TextAnalytics.json:1831]: Use `x-ms-enum` extension.

File: specification/cognitiveservices/data-plane/TextAnalytics/preview/v3.1-preview.4/TextAnalytics.json [hunk 4]
```diff
@@ -1796,25 +1798,67 @@
...
+            "Positive",
+            "Positive Possible",
+            "Neutral Possible",
+            "Negative Possible",
+            "Negative"
+          ]
+        },
+        "association": {
+          "description": "Describes if the entity is the subject of the text or if it describes someone else. Possible values: subject, other",
```
Comment (assafi) [inline specification/cognitiveservices/data-plane/TextAnalytics/preview/v3.1-preview.4/TextAnalytics.json:1840]: nit: Finish descriptions with `.`

Comment (assafi) [reply specification/cognitiveservices/data-plane/TextAnalytics/preview/v3.1-preview.4/TextAnalytics.json:1840]: Also, no need to list the possible values if you use an enum. The values are already listed that way.

File: specification/cognitiveservices/data-plane/TextAnalytics/preview/v3.1-preview.4/TextAnalytics.json [hunk 5]
```diff
@@ -1796,25 +1798,67 @@
...
+    },
+    "HealthcareRelationEntity": {
+      "type": "object",
+      "required": [
+        "ref",
+        "role"
+      ],
+      "properties": {
+        "ref": {
+          "description": "Reference link to the entity object.",
```
Comment (assafi) [inline specification/cognitiveservices/data-plane/TextAnalytics/preview/v3.1-preview.4/TextAnalytics.json:1857]: Add: Using a JSON pointer [RFC 6901](https://tools.ietf.org/html/rfc6901) (URI Fragment Identifier Representation).

File: specification/cognitiveservices/data-plane/TextAnalytics/preview/v3.1-preview.4/TextAnalytics.json [hunk 6]
```diff
@@ -1796,25 +1798,67 @@
...
           "type": "string"
         },
-        "target": {
-          "description": "Reference link to the target entity.",
+        "role": {
+          "description": "Role of entity in the relationship. For example: 'CD20-positive diffuse large B-cell lymphoma' has the following entities with their roles in parenthesis:  CD20 (GeneOrProtein), Positive (Expression), diffuse large B-cell lymphoma (Diagnosis).",
```
Comment (assafi) [inline specification/cognitiveservices/data-plane/TextAnalytics/preview/v3.1-preview.4/TextAnalytics.json:1861]: The common use case is usually simply an `Attribute` and `Entity`, right?

File: specification/cognitiveservices/data-plane/TextAnalytics/preview/v3.1-preview.4/TextAnalytics.json (hunk 1 above)
Comment (assafi) [reply specification/cognitiveservices/data-plane/TextAnalytics/preview/v3.1-preview.4/TextAnalytics.json:1784]: Ignore. Misread the swagger.

File: specification/cognitiveservices/data-plane/TextAnalytics/preview/v3.1-preview.4/TextAnalytics.json (hunk 6 above)
Comment (ctufts) [reply specification/cognitiveservices/data-plane/TextAnalytics/preview/v3.1-preview.4/TextAnalytics.json:1861]: Correct, the most common use case is Attribute and Entity

**Comment by ctufts** [comment]
> > @ctufts is this ready to merge?
> > @tjprescott  Not yet, I'd like to get a review in from some other TA members first. Will update you when it is ready for merging.

@tjprescott I believe this is ready to merge now.

File: specification/cognitiveservices/data-plane/TextAnalytics/preview/v3.1-preview.4/TextAnalytics.json [hunk 7]
```diff
@@ -1790,12 +1790,14 @@
           "$ref": "#/definitions/Entity"
         },
         {
-          "required": [
-            "isNegated"
-          ],
           "properties": {
-            "isNegated": {
-              "type": "boolean"
+            "assertions": {
```
Comment (assafi) [inline specification/cognitiveservices/data-plane/TextAnalytics/preview/v3.1-preview.4/TextAnalytics.json:1794]: This needs to be singular. `assertion`

Comment (ctufts) [reply specification/cognitiveservices/data-plane/TextAnalytics/preview/v3.1-preview.4/TextAnalytics.json]: updated

**Review by abhahn — COMMENTED** [review]
One thing that has come up in discussion with the SDK team is about whether or not we could include the different relation types as an enum in the Swagger.  I this something that is feasible, or are there reasons we shouldn't do this?

**Comment by assafi** [comment]
> One thing that has come up in discussion with the SDK team is about whether or not we could include the different relation types as an enum in the Swagger. I this something that is feasible, or are there reasons we shouldn't do this?

@abhahn This will tie the set of returned types to a specific API version. So new types for example could only be added in a new API version vs. a new `model-version`.

**Pull request merged**